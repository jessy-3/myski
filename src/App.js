import {Header} from './components/Header/Header';
import ResortList from './components/ResortList/ResortList';
import {InfoPanel}  from './components/InfoPanel/InfoPanel';
import {resorts} from './Resorts';
import React, { useState, useEffect } from 'react';

function App() {
  const [action, setAction]=useState("Read");
  const [resort, setResort]=useState();

  useEffect(()=> {
    var index;
    var newresort;
    switch(action) {
      case 'Create':
        if (resort?.id) {
          newresort = {
            id: 0, 
            name: '', 
            location: '', 
            num_skiruns: 1,
            imgsrc: ''
          };
          setResort(newresort);
        } 
        break;
      case 'Created':
        newresort = resort;
        // Retrieve max resort id to assign a unique auto increased id number.
        newresort.id = resorts.reduce((a,b,)=> (parseInt(a.id) > parseInt(b.id) ? parseInt(a.id) : parseInt(b.id))) + 1;
        resorts.push(newresort);
        setAction("Read");
        break;
      case 'Edit':
        // index = resorts.indexOf(resorts.filter((x) => x===resort)[0]);
        break;
      case 'Edited':
        index = resorts.indexOf(resorts.filter((x) => x['id']===resort.id)[0]);
        resorts[index]['name'] = resort.name;
        resorts[index]['location'] = resort.location;
        resorts[index]['num_skiruns'] = parseInt(resort.num_skiruns);
        if (resort.imgsrc)
          resorts[index]['imgsrc'] = resort.imgsrc;
        setAction("Read");
        break;
      case 'Delete':
        index = resorts.indexOf(resorts.filter((x) => x===resort)[0]);
        delete resorts[index];
        setAction("Read");
        break;
      default:
        console.log("Read Mode")
    }
  }, [action, resort])

  return(
    <>
      <Header />
      <ResortList resorts={resorts} setresort={ setResort } setaction={ setAction } />
      <InfoPanel resort={resort} setresort={ setResort } setaction={ setAction } action={action} />
    </>
  );
}

export default App;