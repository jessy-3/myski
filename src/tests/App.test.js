import { render, screen } from '@testing-library/react';
import renderer from "react-test-renderer";
import App from '../App';

describe('App component', () => {
  test('renders the title of the website and main', () => {
    render(<App />);

    const title = screen.getByText(/Popular Ski Resorts/i);
    expect(title).toBeInTheDocument();
    expect(screen.getByRole('main')).toBeInTheDocument();
  });

  //Snapshot tests to make sure UI does not change unexpectedly
  test("it renders a correct snapshot", () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
 })



