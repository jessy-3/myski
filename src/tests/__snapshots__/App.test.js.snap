// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component it renders a correct snapshot 1`] = `
Array [
  <header
    className="header"
  >
    Popular Ski Resorts
  </header>,
  <main
    className="resortList"
    data-testid="resort-list"
  >
    <div
      data-testid="item-container"
    >
      <div
        className="resortItem"
      >
        <div
          className="resort_list_item_title"
        >
          <h3>
            Ski Resort 1
          </h3>
        </div>
        <hr />
        <div
          className="resort_list_item_content"
        >
          <div
            className="resort_list_item_img_wrap"
          >
            <img
              alt="Ski Resort 1"
              src="./images/resort1.jpg"
              title="Ski Resort 1"
              width="100%"
            />
          </div>
          <div
            className="resort_list_item_text"
          >
            <table
              className="infor_table"
            >
              <tbody>
                <tr>
                  <td>
                    Name: 
                    Ski Resort 1
                  </td>
                </tr>
                <tr>
                  <td>
                    Location: 
                    Location 1
                  </td>
                </tr>
                <tr>
                  <td>
                    Number of ski runs: 
                    11
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
        <div
          className="resort_list_item_btn"
        >
          <button
            data-testid="edit"
            onClick={[Function]}
          >
            Edit
          </button>
          <button
            data-testid="delete"
            onClick={[Function]}
          >
            Delete
          </button>
        </div>
      </div>
    </div>
    <div
      data-testid="item-container"
    >
      <div
        className="resortItem"
      >
        <div
          className="resort_list_item_title"
        >
          <h3>
            Ski Resort 22
          </h3>
        </div>
        <hr />
        <div
          className="resort_list_item_content"
        >
          <div
            className="resort_list_item_img_wrap"
          >
            <img
              alt="Ski Resort 22"
              src="./images/resort2.webp"
              title="Ski Resort 22"
              width="100%"
            />
          </div>
          <div
            className="resort_list_item_text"
          >
            <table
              className="infor_table"
            >
              <tbody>
                <tr>
                  <td>
                    Name: 
                    Ski Resort 22
                  </td>
                </tr>
                <tr>
                  <td>
                    Location: 
                    Location 2
                  </td>
                </tr>
                <tr>
                  <td>
                    Number of ski runs: 
                    12
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
        <div
          className="resort_list_item_btn"
        >
          <button
            data-testid="edit"
            onClick={[Function]}
          >
            Edit
          </button>
          <button
            data-testid="delete"
            onClick={[Function]}
          >
            Delete
          </button>
        </div>
      </div>
    </div>
    <div
      data-testid="item-container"
    >
      <div
        className="resortItem"
      >
        <div
          className="resort_list_item_title"
        >
          <h3>
            Ski Resort 333
          </h3>
        </div>
        <hr />
        <div
          className="resort_list_item_content"
        >
          <div
            className="resort_list_item_img_wrap"
          >
            <img
              alt="Ski Resort 333"
              src="./images/resort3.webp"
              title="Ski Resort 333"
              width="100%"
            />
          </div>
          <div
            className="resort_list_item_text"
          >
            <table
              className="infor_table"
            >
              <tbody>
                <tr>
                  <td>
                    Name: 
                    Ski Resort 333
                  </td>
                </tr>
                <tr>
                  <td>
                    Location: 
                    Location 3
                  </td>
                </tr>
                <tr>
                  <td>
                    Number of ski runs: 
                    13
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
        <div
          className="resort_list_item_btn"
        >
          <button
            data-testid="edit"
            onClick={[Function]}
          >
            Edit
          </button>
          <button
            data-testid="delete"
            onClick={[Function]}
          >
            Delete
          </button>
        </div>
      </div>
    </div>
    <div
      data-testid="item-container"
    >
      <div
        className="resortItem"
      >
        <div
          className="resort_list_item_title"
        >
          <h3>
            Ski Resort 4
          </h3>
        </div>
        <hr />
        <div
          className="resort_list_item_content"
        >
          <div
            className="resort_list_item_img_wrap"
          >
            <img
              alt="Ski Resort 4"
              src="./images/resort4.webp"
              title="Ski Resort 4"
              width="100%"
            />
          </div>
          <div
            className="resort_list_item_text"
          >
            <table
              className="infor_table"
            >
              <tbody>
                <tr>
                  <td>
                    Name: 
                    Ski Resort 4
                  </td>
                </tr>
                <tr>
                  <td>
                    Location: 
                    Location 4
                  </td>
                </tr>
                <tr>
                  <td>
                    Number of ski runs: 
                    14
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
        <div
          className="resort_list_item_btn"
        >
          <button
            data-testid="edit"
            onClick={[Function]}
          >
            Edit
          </button>
          <button
            data-testid="delete"
            onClick={[Function]}
          >
            Delete
          </button>
        </div>
      </div>
    </div>
    <div
      data-testid="item-container"
    >
      <div
        className="resortItem"
      >
        <div
          className="resort_list_item_title"
        >
          <h3>
            Ski Resort 5
          </h3>
        </div>
        <hr />
        <div
          className="resort_list_item_content"
        >
          <div
            className="resort_list_item_img_wrap"
          >
            <img
              alt="Ski Resort 5"
              src="./images/resort5.webp"
              title="Ski Resort 5"
              width="100%"
            />
          </div>
          <div
            className="resort_list_item_text"
          >
            <table
              className="infor_table"
            >
              <tbody>
                <tr>
                  <td>
                    Name: 
                    Ski Resort 5
                  </td>
                </tr>
                <tr>
                  <td>
                    Location: 
                    Location 5
                  </td>
                </tr>
                <tr>
                  <td>
                    Number of ski runs: 
                    15
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
        <div
          className="resort_list_item_btn"
        >
          <button
            data-testid="edit"
            onClick={[Function]}
          >
            Edit
          </button>
          <button
            data-testid="delete"
            onClick={[Function]}
          >
            Delete
          </button>
        </div>
      </div>
    </div>
  </main>,
  <aside
    className="panel"
  >
    <div
      className="panel_item_btn"
    >
      <button
        onClick={[Function]}
      >
        Add New
      </button>
    </div>
  </aside>,
]
`;
